src/App.tsx

```diff
@@ -1,26 +1,293 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
 import React from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID } from './api/todos';
+import { getTodos } from './api/todos';
+import { useEffect, useState } from 'react';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Todo';
+import { ErrorNotification } from './components/error/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  // const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [filter, setFilter] = useState<Filter>('all');
+  const [isAdding, setIsAdding] = useState(false);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const activeCount = todos.filter(todo => !todo.completed).length;
+  const allCompleted = todos.length > 0 && todos.every(todo => todo.completed);
+  const hasCompleted = todos.some(todo => todo.completed);
+  const API_URL = 'https://mate.academy/students-api';
+
+  useEffect(() => {
+    // setIsLoading(true);
+    setError('');
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => setError('Unable to load todos'));
+    // .finally(() => setIsLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      const timer = setTimeout(() => {
+        setError('');
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [error]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const filteredTodos = todos.filter(todo => {
+    if (filter === 'active') {
+      return !todo.completed;
+    }
+
+    if (filter === 'completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  const handleCloseError = () => {
+    setError('');
+  };
+
+  const handleAddTodo = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = title.trim();
+
+    if (!trimmedTitle) {
+      setError('Title should not be empty');
+
+      return;
+    }
+
+    setIsAdding(true);
+
+    const newTempTodo: Todo = {
+      id: 0,
+      userId: USER_ID,
+      title: trimmedTitle,
+      completed: false,
+    };
+
+    setTempTodo(newTempTodo);
+
+    try {
+      const response = await fetch(`${API_URL}/todos`, {
+        method: 'POST',
+        body: JSON.stringify({
+          userId: USER_ID,
+          title: trimmedTitle,
+          completed: false,
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
```

Comment:
Create helper functions which will get, delete, update, create todo. Move them to separate file